src/csprojWriter.ts

```diff
@@ -108,15 +118,21 @@ export class CsProjWriter {
         let items: Array<Object> = Object(parsedXml).Project.ItemGroup;
 
         for (let item of items) {
-            let actions: Array<Object> = Object.keys(item).map(key => Object(item)[key])[0];
-            for (let action of actions) {
-                let include: string = Object(action)["$"].Include;
-                if (include === itemPath || (isDir && include.startsWith(itemPath))) {
-                    actions.splice(actions.indexOf(action));
+            let actionsArray: Array<Array<Object>> = Object.keys(item).map(key => Object(item)[key]);
+            
+            for (let index = 0; index < actionsArray.length; index++) {
+                let actions = actionsArray[index];
+                for (let action = 0; action < actions.length; action++) {
+                    let include: string = Object(actions[action])["$"].Include;
+                    if (include === itemPath || (isDir && include.startsWith(itemPath))) {
+                        actions.splice(action, 1);
+                    }
                 }
+
+                if (actions.length === 0) actionsArray.splice(actionsArray.indexOf(actions), 1);
             }
 
-            if (actions.length === 0) items.splice(items.indexOf(item));
+            if (actionsArray.length === 0) items.splice(items.indexOf(item), 1);
         }
```

Comment:
Use same algorithm as with `getItemGroupByPath` to fix bugs when removing items from `ItemGroups` that contain several different `BuildActions`